pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -309,6 +364,38 @@ def get_sweep_points(self):
         if self.sp is not None:
             self.sp = SweepPoints(self.sp)
 
+    def get_cal_points(self):
+        cal_points = self.get_param_value('cal_points')
+        last_ge_pulses = self.get_param_value('last_ge_pulses',
+                                              default_value=False)
+
+        cal_states_rotations = {qbn: [] for qbn in self.qb_names}
+        try:
+            self.cp = CalibrationPoints.from_string(cal_points)
+            # for now assuming the same for all qubits.
+            self.cal_states_dict = self.cp.get_indices(
+                self.qb_names)[self.qb_names[0]]
+            cal_states_rots = self.cp.get_rotations(
+                last_ge_pulses, self.qb_names[0])[self.qb_names[0]] if \
+                self.rotate else cal_states_rotations
+            self.cal_states_rotations = self.get_param_value(
+                'cal_states_rotations', default_value=cal_states_rots)
+        except TypeError as e:
+            log.error(e)
+            log.warning("Failed retrieving cal point objects or states. "
+                        "Please update measurement to provide cal point object "
+                        "in metadata. Trying to get them using the old way ...")
+            self.cal_states_rotations = self.get_param_value(
+                'cal_states_rotations', default_value=cal_states_rotations) \
+                if self.rotate else cal_states_rotations
+            self.cal_states_dict = self.get_param_value('cal_states_dict',
+                                                        default_value={})
+        if self.cal_states_rotations is None:
```

Comment:
Not really, it can be that somehow it was None in the metadata. This line essentially gives this attribute an expected default value. We could raise a warning here to let the user know None is not accepted.

Reply:
if self.cal_states_rotations is None:
    # if an error happens above or None was provided in the metadata

I can't judge whether it is a common thing that the user writes None into the metadata and whether we need to warn them about it. I just wanted a comment to make clear for a reader of the code why we end up in this branch.